### ISSUE TITLE
Multi-Select ErrorAnnotation bug

### ISSUE THREAD

**Issue by rmadsen-ks** [opentap/opentap#1700]
An exception is sometimes thrown when getting the error of multi-selecting objects.

Reproduced by this unit-test:
```
        public class EmbedObject
        {
            public Verdict C { get; set; }
        }

        public class EmbedeeObject : ValidatingObject
        {
            [EmbedProperties]
            public EmbedObject Embed { get; set; } = new EmbedObject();
        }

        [Test]
        public void EmbedMultiSelect()
        {
            var aobj = new EmbedeeObject();
            var bobj = new EmbedeeObject();
            var selection = new object[]
            {
                aobj, bobj
            };

            var a = AnnotationCollection.Annotate(selection);
            foreach (var member in a.Get<IMembersAnnotation>().Members)
            {
                var errs = member.Get<IErrorAnnotation>()?.Errors.ToArray();
                Assert.AreEqual(0, errs.Length);
            }
        }
```

**Comment by rmadsen-ks** [comment]
**Testing**

Have two test steps with the same validation error. Multi-select those. Verify that the error still shows and there is no unexpected error messages in the log.

**Comment by sebastian-pop** [comment]
Reproduced on 9.26

![Image](https://github.com/user-attachments/assets/959f15f7-ae3e-42bf-b22f-3da97c578274)

Confirmed as fixed on 9.27

![Image](https://github.com/user-attachments/assets/88e8b36b-ee1b-4923-a32f-81f78007ea64)
